### PR TITLE
TomThumb font now compressed to fix bounding box text width problem

### PR DESCRIPTION
**Pull request by BlueAndi**
Bitmaps now compressed (like the other fonts), to fix the bounding box problem, because non-compressed the calculated text width were wrong. Additional advantage is that it needs less program space.

#266

### CODE REVIEW

File: Fonts/TomThumb.h [hunk 1]
```diff
@@ -43,429 +43,435 @@
 ** https://gist.github.com/skelliam/322d421f028545f16f6d
 ** William Skellenger (williamj@skellenger.net)
 ** Twitter: @skelliam
-**
+** 
+** Jan 09, 2020: Bitmaps now compressed, to fix the bounding box problem,
```
Comment (BillyDonahue) [inline Fonts/TomThumb.h]: Can you attach the code you used to generate the font header?
It's good to be able to generate the header from the upstream font source BDF file.
We previously had a link to the @skelliam bdf2adafruit.py gist.

I would like to suggest that the @skelliam bdf2adafruit.py be modified or rewritten to produce your improved header format, and then checked into the fontconvert/ directory here, so we never lose access to it.

File: Fonts/TomThumb.h [hunk 2]
```diff
@@ -28,13 +28,13 @@
 ** THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
 ** (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF
 ** THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
-**
-** Modifications to Tom Thumb for improved readability are from Robey Pointer,
+**
```
[the remark this answers is not in the document]
Comment (BlueAndi) [reply Fonts/TomThumb.h]: *autsch* I wondered why clang check failed.

File: Fonts/TomThumb.h (hunk 1 above)
Comment (BlueAndi) [reply Fonts/TomThumb.h]: I changed the bdf2adafruit.py gist a little bit, to generate the compressed set. ;-)
Will commit it to /fontconvert folder.

**Comment by BlueAndi** [comment]
@BillyDonahue The CI checks still fails (clang part), but I don't know why. Can you give me a hint?

**Comment by BillyDonahue** [comment]
@BlueAndi   If you expand the details on the failed "clang" check, it shows a diff. That's showing the difference between what clang-format wants and what you have. You can clean that up and apply it with `patch` directly.

https://github.com/adafruit/Adafruit-GFX-Library/pull/268/checks

...
2020-01-09T18:29:47.8803821Z --- ./Fonts/TomThumb.h	(original)
2020-01-09T18:29:47.8805049Z +++ ./Fonts/TomThumb.h	(reformatted)

Looks like it wants slightly different padding for the end-of-line comments.
And it wants to bin-pack the fields of the TomThumb GFXFont.

    -const GFXfont TomThumb PROGMEM = {
    -    (uint8_t  *)TomThumbBitmaps,
    -    (GFXglyph *)TomThumbGlyphs,
    -    0x20, 0x7E, 6 };
    +const GFXfont TomThumb PROGMEM = {(uint8_t *)TomThumbBitmaps,
    +                                  (GFXglyph *)TomThumbGlyphs, 0x20, 0x7E, 6};

The easiest way to fix this might be to download clang-format 3.8.0 and run it yourself.

**Comment by BlueAndi** [comment]
@BillyDonahue Yeah, I did it. Is it possible that the CI clang output explains it better? At the end it only shows error. Thats it.

**Comment by PaintYourDragon** [comment]
Thank you!

**Pull request merged**